### PR TITLE
Api doc (relates to #667)

### PR DESCRIPTION
**Pull request by bjoernma**
Added first version of an API (mainly its usage) documentation as md file. Should solve issue #667 .

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# OpenRailwayMap-API
+
+ OpenRailwayMap as an online service has different channels from which information can be gathered. One is the [OpenRailwayMap website](https://www.openrailwaymap.org/) to visit the map, another is the [OpenRailwayMap-API](https://api.openrailwaymap.) to retrieve INFORMATION about facilities, milestone positions or the network length for different railway infrastucture operators.
```
Comment (DerDakon) [inline API.md:3]: The is "org" missing in the second URL.

File: API.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ The API uses JavaScript to extract the information brought in the URL. Depending on the request type (MODULE in the example) it loads the module js script. For further information visit the source code on [Github](https://github.com/OpenRailwayMap/OpenRailwayMap/).
+
+---
+
+## Use
+
+[https://api.openrailwaymap.org/MODULE?QUERY](https://api.openrailwaymap.org/networklength)
+
+ Examples: To receive information about the station Hamburg Hbf (main station) in Germany you can type:
+[https://api.openrailwaymap.org/facility?name=Hamburg Hbf](https://api.openrailwaymap.org/facility?name=Hamburg Hbf)
```
Comment (DerDakon) [inline API.md:16]: The spaces should probably be turned into '+' or '%20'.

File: API.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+> [{"lat":10.0075644,"lon":53.5526958991248,"name":"Hamburg Hauptbahnhof","uicname":"Hamburg Hbf (S-Bahn)","uicref":"8098549","ref":"AHS","id":"2459919677","type":"station","operator":"S-Bahn Hamburg GmbH","stationcategory":"1"},{"lat":10.0064364,"lon":53.5531992991249,"name":"Hamburg Hauptbahnhof","uicname":"Hamburg Hbf","uicref":"8002549","ref":"AH","id":"1680910488","type":"station","operator":"DB Station&Service AG","stationcategory":"1"}]
+
+If using both e.g. name and ref, the api uses the first of the keywords and ignores the second. The operator keyword may be placed at any point.
+
+---
+
+### Milestone module
+
+The milestone keyword returns the position of a milestone on a line (with Tag ref) which is operated by an operator. Missing milestones are interpolated.
```
Comment (DerDakon) [inline API.md:52]: This will not only return milestones, but any object tagged with a distance on the track, e.g. signals and crossings.

**Review by DerDakon — CHANGES_REQUESTED** [review]

File: API.md (hunk 2 above)
Comment (bjoernma) [reply API.md:16]: Both work in the URL (in my setup), but it's changed

File: API.md (hunk 1 above)
Comment (bjoernma) [reply API.md]: Changed, thanks!

File: API.md (hunk 3 above)
Comment (bjoernma) [reply API.md]: Thanks, it's changed in both milestone.js as well as this description

**Review by DerDakon — APPROVED** [review]
LGTM, but @Nakaner knows better about these code.

**Comment by bjoernma** [comment]
just rebased to current master (and found a missing ")" beforehand).

**Review by Nakaner — CHANGES_REQUESTED** [review]

**Comment by Nakaner** [comment]
I used parts of your pull request to improve the [documentation of the new API](https://github.com/OpenRailwayMap/OpenRailwayMap-api/blob/master/README.md#api) which went live yesterday. Reading your text carefully helped me to find bugs and describe precisely what the new API searches for (and what not).

**Pull request closed without merging**